augurpy/calculate_AUC.py

```diff
@@ -1,66 +1,70 @@
-    augur_mode: str = "default",
-    classifier: str = "rf",
-    rf_params: Dict = None,
-    lr_params: List = None,
-) -> Tuple[Union[Dict, AnnData]]:
-    """Calculates the Area under the Curve using the given classifier.
+
+def cross_validate_subsample(
+    input: AnnData,
+    estimator: Union[RandomForestRegressor, RandomForestClassifier, LogisticRegression],
+    augur_mode: str,
+    subsample_size: int,
+    folds: int,
+    feature_perc: float,
+    subsample_idx: int,
+    random_state: Optional[int],
+) -> DataFrame:
+    """Cross validate subsample.
 
     Args:
-        input: Anndata or matrix containing gene expression values (cells in rows, genes in columns) and optionally meta
-            data about each cell.
-        meta: Optional Pandas DataFrame containing meta data about each cell.
-        label_col: column of the meta data or the Anndata or matrix containing the condition labels for each cell
-            in the cell-by-gene expression matrix
-        cell_type_col: column of the meta DataFrame or the Anndata or matrix containing the cell type labels for each
-            cell in the cell-by-gene expression matrix
-        n_subsamples: number of random subsamples to draw from complete dataset for each cell type
-        subsample_size: number of cells to subsample randomly per type from each experimental condition
-        folds: number of folds to run cross validation on
-        min_cells: minimum number of cells for a particular cell type in each condition in order to retain that type for
-            analysis (depricated..)
-        var_quantile: quantile of highly variable genes to retain for each cell type using the variable gene filter
-        feature_perc: proportion of genes that are randomly selected as features for input to the classifier in each
-            subsample using the random gene filter
-        n_threads: number of threads to use for parallelization
-        show_progress: if `True` display a progress bar for the analysis with estimated time remaining
+        input: Pandas DataFrame
+        estimator: classifier
         augur_mode: one of default, velocity or permute. Setting augur_mode = "velocity" disables feature selection,
             assuming feature selection has been performed by the RNA velocity procedure to produce the input matrix,
             while setting augur_mode = "permute" will generate a null distribution of AUCs for each cell type by
             permuting the labels
-        classifier: classifier to use in calculating the area under the curve either random forest or logistic regression
-        rf_params: list of parameters for random forest
-        lr_params: list of parameters for logistic regression
+        subsample_size: number of cells to subsample randomly per type from each experimental condition
+        folds: number of folds to run cross validation on
+        subsample_idx: index of the subsample
+        random_state: set random seed
```

Comment:
What does the random seed effect? Very briefly mention here. Note if it does not affect something it might (e.g. Numpy seed or whatever)